src/main/java/ch/njol/skript/expressions/ExprIndices.java

```diff
@@ -101,6 +101,7 @@ protected String[] get(Event e) {
 		if (sort) {
 			int direction = descending ? -1 : 1;
 			return variable.entrySet().stream()
+				.peek((entry) -> entry.setValue(entry.getValue() instanceof Map<?,?> ? ((Map<?, ?>) entry.getValue()).get(null) : entry.getValue()))
```

Comment:
Due to the usage of `setValue` here, this has the potential to make modifications to the variable map. You'll probably need to instead map into a new Pair to avoid modifying the backing map.

Reply:
See `assert {_test::3::*} is 6 with "modified children wrongly"` as an example test

Reply:
the suggested test worked fine, should it still be mapped to a new pair just in case?

Reply:
Yep. Additionally, your test appears to be wrong, as you have `assert {_test::3::*} is NOT 6` when it actually should be (get rid of `not`)

Reply:
The test failed for me, I'll investigate